### PR TITLE
bring in AL fork into here to have a single version.

### PR DESCRIPTION
**Pull request by motobob**
slightly reduce dependancy hell by merging together with AL fork here and use only `erlcloud/lhttpC` in future

### CODE REVIEW

File: test/lhttpc_manager_tests.erl [hunk 1]
```diff
@@ -300,13 +298,27 @@ closed_race_cond() ->
     ManagerPid = whereis(lhttpc_manager),
     true = erlang:suspend_process(ManagerPid),
 
+    case CloseBeforeDispatch of
+        true ->
+            gen_tcp:close(LS), % a closed message should be sent to the manager
+            timer:sleep(1000);
```
Comment (kkuzmin) [inline test/lhttpc_manager_tests.erl]: why sleep for 1 sec is need here?

**Comment by kkuzmin** [comment]
Please fix failed checks.
+1 for bringing it up to speed.

File: test/lhttpc_manager_tests.erl (hunk 1 above)
Comment (kkuzmin) [inline test/lhttpc_manager_tests.erl]: Sorry, I didn't notice it is in tests.

**Comment by nalundgaard** [comment]
@motobob do these tests pass for you locally? Glancing the error output from travis, I suspect this is exactly what I was seeing when I toyed with unifying erlcloud/alertlogic lhttpc a month or two ago.

If that can be resolved, I am all for it.

**Comment by motobob** [comment]
@nalundgaard they pass from alertlogic/master but they fail locally with the same error if merged together. it's all about canonical headers which is partially addressed by @fp at #4 but there is still one test failing. i'll merge it to a branch and fix it all to send send a mergable PR.

**Pull request merged**